proxy/README.md

```diff
@@ -0,0 +1,91 @@
+# `cc-proxy`
+
+`cc-proxy` is a daemon offering access to the
+[`hyperstart`](https://github.com/hyperhq/hyperstart) VM agent to multiple
+clients on the host.
+
+![High-level Architecture Diagram](../documentation/high-level-overview.png)
+
+- The `hyperstart` interface consists of:
+    - A control channel on which the [`hyperstart` API]
+      (https://github.com/hyperhq/runv/tree/master/hyperstart/api/json) is
+      delivered.
+    - An I/O channel with the stdin/stout/stderr streams of the processes
+      running inside the VM multiplexed onto.
+- `cc-proxy`'s main role is to:
+    - Arbitrate access to the `hyperstart` control channel between all the
+      instances of the OCI runtimes and `cc-shim`.
+    - Route the I/O streams between the various shim instances and `hyperstart`.
```

Comment:
So the idea is that the cc-shim will read IO streams and shim will be responsible to write to the correct place the tty output. As well that shim will take container input and ask to the proxy to send it to hyperstart -> process